packages/core/src/baseclient.ts

```diff
@@ -650,6 +650,11 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
 
     this.emit('preprocessEvent', event, hint);
 
+    const eventId = event.event_id || hint.event_id;
+    if (eventId) {
+      isolationScope.setLastEventId(eventId);
+    }
```

Comment:
m: I think we pass all kinds of events through `_prepareEvent` right? Before removing this function, we only updated the last event id for error events. I'd argue we should do the same now. 

see https://github.com/getsentry/sentry-javascript/blob/90b45f06c9765b899dc68a8ac828b9816f86cb82/packages/core/src/hub.ts#L396-L398

Reply:
Sorry, I think I worded this incorrectly. I'm totally fine with error and message events that's all good. They all have `event.type === 'undefined'`. 

As for Replay or transaction/span events, these also go through `_prepareEvent` but they have a different `event.type`. I woudn't update lastEventId for them because

1. then users will much more likely get completely unexpected `lastEventId` return values. For example, a transaction could end right between the error being captured and users calling `lastEventId` --> the return value would return the transaction event id.
2. previous behaviour was also to not update the id for these events. 

I agree, naming-wise `lastEventId` implies any event but this has historic reasons. This API probably existed long before we decided to send spans/transactions and also treat them as events with a different type.

Happy to be overruled but I think we should consider at least point 1

Reply:
I agree with lukas, that makes the most sense to me too! :)

Reply:
@Lms24 fixed. Thanks for the explanation, I assumed too much what an Event is :)